tango/local_workspace.py

```diff
@@ -277,9 +277,42 @@ def work_dir(self, step: Step) -> Path:
         result.mkdir(parents=True, exist_ok=True)
         return result
 
-    @staticmethod
-    def dir_is_empty(dir: Path):
-        return not any(True for _ in dir.iterdir())
+    @classmethod
+    def guess_step_dir_state(cls, dir: Path) -> Set[StepState]:
+        """
+        Returns the possible states of a given step dir, to the best of our knowledge.
+
+        :param dir: the step dir to example
+        :return: a set of possible states for the step
+        """
+
+        # If the directory doesn't exist, the step is incomplete.
+        if not dir.exists():
+            return {StepState.INCOMPLETE}
+
+        # If the lock file exists and is locked, the step is running.
+        lock_file = dir / "lock"
+        if lock_file.exists():
+            lock = FileLock(lock_file)
+            try:
+                lock.acquire(0.1)
+                lock.release()
+            except TimeoutError:
+                return {StepState.RUNNING}
```

Comment:
@epwalsh, is this reliable? If the lock file isn't locked, can we safely assume that the step can't possibly be running?

I could swear I saw this fail on the Mac filesystem, but I can't reproduce any problems with it now.

Reply:
I think that's safe to assume